src/amp-story-player/amp-story-entry-point/amp-story-entry-point-impl.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Source for this constant is css/amp-story-entry-point.css
+import {cssText} from '../../build/amp-story-entry-point.css';
+
+/**
+ * <amp-story-entry-point> component for embedding stories and launching them in
+ * the <amp-story-player>.
+ *
+ * Note that this is a vanilla JavaScript class and should not depend on AMP
+ * services, as v0.js is not expected to be loaded in this context.
+ */
+export class AmpStoryEntryPoint {
+  /**
+   * @param {!Window} win
+   * @param {!Element} element
+   * @constructor
+   */
+  constructor(win, element) {
+    /** @private {!Window} */
+    this.win_ = win;
+
+    /** @private {!Element} */
+    this.element_ = element;
+
+    /** @private {!Document} */
+    this.doc_ = this.win_.document;
+
+    /** @private {boolean} */
+    this.isBuilt_ = false;
+
+    /** @private {boolean} */
+    this.isLaidOut_ = false;
+
+    /** @private {?Element} */
+    this.rootEl_ = null;
+  }
+
+  /** @public */
+  buildCallback() {
+    if (this.isBuilt_) {
+      return;
+    }
+
+    this.initializeShadowRoot_();
+
+    this.isBuilt_ = true;
+  }
+
+  /** @public */
+  layoutCallback() {
+    if (this.isLaidOut_) {
+      return;
+    }
+
+    this.isLaidOut_ = true;
+  }
+
+  /** @private */
+  initializeShadowRoot_() {
+    this.rootEl_ = this.doc_.createElement('main');
+
+    // Create shadow root
+    const shadowRoot = this.element_.attachShadow({mode: 'open'});
```

Comment:
This will fail on many browsers, we should have a fallback.

Reply:
I'll address as a follow-up since this is something that we also do for the player.

Reply:
Sounds good, is is tracked somewhere?

Reply:
https://github.com/ampproject/amphtml/issues/27358